File: services/user-service/routes/userRoutes.js
// services/user-service/routes/userRoutes.js
const express = require('express');
const {
    signup,
    login,
    getUserProfile,
    emailVerification,
    updateProfilePicture,
    updatePassword,
    updateEmail
} = require('../controllers/userController');
const router = express.Router();
const multer = require('multer');

const upload = multer({ storage: multer.memoryStorage() });

router.post('/signup', signup);
router.post('/login', login);
router.get('/getinfo', getUserProfile);
router.post('/emailverification', emailVerification);
router.post('/update-profile-picture', upload.single('file'), updateProfilePicture);
router.patch('/update-password', updatePassword); // Route for updating password
router.patch('/update-email', updateEmail);       // Route for updating email

module.exports = router;

File: services/user-service/routes/userRoutes.py
# routes/user_routes.py
from flask import Blueprint
from controllers.user_controller import (
    signup,
    login,
    get_user_profile,
    email_verification,
    update_profile_picture,
    update_password,
    update_email
)
from flask_uploads import UploadSet, IMAGES, configure_uploads

user_routes = Blueprint('user_routes', __name__)

# Configure file uploads
photos = UploadSet('photos', IMAGES)


# Routes
user_routes.route('/signup', methods=['POST'])(signup)
user_routes.route('/login', methods=['POST'])(login)
user_routes.route('/getinfo', methods=['GET'])(get_user_profile)
user_routes.route('/emailverification', methods=['POST'])(email_verification)


user_routes.route('/update-profile-picture', methods=['POST'])(update_profile_picture)

user_routes.route('/update-password', methods=['PATCH'])(update_password)
user_routes.route('/update-email', methods=['PATCH'])(update_email)
